src/nvim/api/command.c

```diff
@@ -650,10 +650,20 @@ String nvim_cmd(uint64_t channel_id, Dict(cmd) *cmd, Dict(cmd_opts) *opts, Error
     }
   }
 
-  // Finally, build the command line string that will be stored inside ea.cmdlinep.
+  // Finally, build the command line string (if it is required) and store in ea.cmdlinep.
   // This also sets the values of ea.cmd, ea.arg, ea.args and ea.arglens.
-  build_cmdline_str(&cmdline, &ea, &cmdinfo, args);
-  ea.cmdlinep = &cmdline;
+  // If command line string isn't required, then just set ea.argc, ea.args and ea.arglens
+  if ((ea.argt & EX_OPTCMDSTR) > 0 || (ea.argt & EX_EXTRA) == 0) {
```

Comment:
Usually `!= 0` is used for this kind of things.
```suggestion
  if ((ea.argt & EX_OPTCMDSTR) != 0 || (ea.argt & EX_EXTRA) == 0) {
```